app/components/ui/molecules/PeoplePicker.js

```diff
@@ -0,0 +1,159 @@
+    return <SuccessMessage>Maximum {maxSelection} choices</SuccessMessage>
+  }
+  return null
+}
+
+const PeoplePickerBody = ({
+  isSelected,
+  maxSelection,
+  minSelection,
+  onCancel,
+  onSubmit,
+  people,
+  selection,
+  toggleSelection,
+}) => (
+  <React.Fragment>
+    <Flex justifyContent="space-between">
+      <SelectedItems onCloseClick={toggleSelection} selection={selection} />
+      <SelectionHint
+        maxSelection={maxSelection}
+        minSelection={minSelection}
+        selection={selection}
+      />
+    </Flex>
+    <Flex flexWrap="wrap" mx={-2}>
+      {people.map(person => (
+        <Box key={person.id} p={2} width={1 / 2}>
+          <PersonPod.ChosenPersonPod
+            iconState={isSelected(person) ? 'selected' : 'add'}
+            institution={person.institution}
+            isIconClickable={
+              isSelected(person) || selection.length < maxSelection
+            }
+            name={person.name}
+            onIconClick={() => toggleSelection(person)}
+          />
+        </Box>
+      ))}
+    </Flex>
+  </React.Fragment>
+)
+
+const PeoplePickerButtons = ({ isValid, onCancel, onSubmit }) => (
+  <Flex>
+    <Box mr={3}>
+      <Button onClick={onCancel}>Cancel</Button>
+    </Box>
+    <Box>
+      <Button disabled={!isValid} onClick={onSubmit} primary>
+        Add
+      </Button>
+    </Box>
+  </Flex>
+)
+
+class PeoplePicker extends React.Component {
+  state = { selection: [] }
+
+  toggleSelection(person) {
+    if (this.isSelected(person)) {
+      this.setState({
+        selection: this.state.selection.filter(p => p.id !== person.id),
+      })
+    } else {
+      this.setState({ selection: this.state.selection.concat(person) })
+    }
+  }
+
+  handleSubmit() {
+    if (this.isValid()) {
+      this.props.onSubmit(this.state.selection)
+    }
+  }
+
+  isSelected(person) {
+    return this.state.selection.some(p => p.id === person.id)
+  }
+
+  isValid() {
+    const selectionLength = this.state.selection.length
+    return (
+      selectionLength >= this.props.minSelection &&
+      selectionLength <= this.props.maxSelection
+    )
+  }
+
+  render() {
+    return this.props.children({
+      ...this.props,
+      isSelected: person => this.isSelected(person),
+      isValid: this.isValid(),
+      selection: this.state.selection,
+      onSubmit: () => this.handleSubmit(),
+      toggleSelection: person => this.toggleSelection(person),
+    })
+  }
+}
+
+PeoplePicker.propTypes = {
+  minSelection: PropTypes.number,
+  maxSelection: PropTypes.number,
+  onCancel: PropTypes.func.isRequired,
+  onSubmit: PropTypes.func.isRequired,
+  people: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
+      name: PropTypes.string.isRequired,
+    }),
+  ).isRequired,
+}
+
+PeoplePicker.defaultProps = {
+  minSelection: 0,
+  maxSelection: Infinity,
+}
+
+PeoplePicker.Body = PeoplePickerBody
```

Comment:
It's probably good to specify the prop types for these sub components too, right? After all, we're giving people a way of rendering them, separately from the `PeoplePicker` alone

Reply:
Probably wouldn't hurt although if you look at the examples, you're expected to spread the argument from the render prop function into these components. Doing that allows the user to not care what those props are and us (the authors) to change the API in future without breaking most people's usages. I've update the docs to mention this.

Reply:
Sure :+1: 

I was imagining that we were exposing `PeoplePicker.Body`, separately from `PeoplePicker`, so that someone could use the body by itself if they wanted. But I see now that the separation is purely because of `grid-styled` - so that we can put `Flex` & `Box` elements between `PeoplePicker` & its children?

We're not actually expecting/suggesting that anyone uses `PeoplePicker.Body` independently from `PeoplePicker`. Makes more sense now. Thanks.